crates/re_query_cache/src/range.rs

```diff
@@ -299,20 +299,56 @@ macro_rules! impl_query_archetype_range {
                     upsert_results::<A, $($pov,)+ $($comp,)*>(arch_views, &mut range_cache.per_data_time)?;
                 }
 
+                Ok(())
+            };
+
+            let iter_callback = |query: &RangeQuery, range_cache: &crate::RangeCache, f: &mut F| -> crate::Result<()> {
+                re_tracing::profile_scope!("range", format!("{query:?}"));
+
+                // We don't bother implementing the slow path here (busy write lock), as that would
+                // require adding a bunch more complexity in order to know whether a range query is
+                // already cached (how can you know whether `TimeInt::MAX` is cached? you need to
+                // clamp queries based on store metadata first, etc).
+                //
+                // We can add the extra complexity if this proves to be glitchy in real-world
+                // scenarios -- otherwise all of this is giant hack meant to go away anyhow.
+
+                // NOTE: Same logic as what the store does.
+                if query.range.min <= TimeInt::MIN {
+                    let mut reduced_query = query.clone();
+                    // This is the reduced query corresponding to the timeless part of the data.
+                    // It is inclusive and so it will yield `MIN..=MIN` = `[MIN]`.
+                    reduced_query.range.max = TimeInt::MIN; // inclusive
```

Comment:
I'm a bit confused by this, why was it not needed before and why ca  the query be malformed to begin with

Reply:
thanks for the link!